tests/experimental/datasources/test_postgres_datasource.py

```diff
@@ -647,6 +649,8 @@ def test_data_source_yaml_has_properties(create_source: CreateSourceFixture):
         )
         source_str = source.__str__()
         assert "order_by:" in source_str
+        # type should be in dumped str even if not explicitly set
+        assert f"type: {asset.type}" in source_str
```

Comment:
Note. These 3 `test_postgres_datasource.py` tests are the only ones that I could actually get to catch the error.
This only happens when an asset is created in code and no `type` field is set.

I couldn't reproduce it with any of the config tests.